lib/fs_utils/source_file.js

```diff
@@ -53,6 +54,15 @@ const updateMap = (path, compiled, wrapped, sourceMap) => {
   if (suffix) node.add(suffix);
   node.source = path;
   node.setSourceContent(path, wrapperContent);
+
+  // the supplied source map might contain more than one source file
+  if (sourceMap && sourceMap.sources) {
+    sourceMap.sources.forEach(path => {
+      const file = fileList.find(path);
+      const content = file && file.source || fs.readFileSync(path).toString();
```

Comment:
sometimes a file won't be on the file list (e.g. if it's `_something.sass` and `_*` files are ignored) so it's probably ok to read these synchronously